common/src/main/java/io/netty/util/ThreadDeathWatcher.java

```diff
@@ -121,10 +125,10 @@ public static boolean awaitInactivity(long timeout, TimeUnit unit) throws Interr
             throw new NullPointerException("unit");
         }
 
-        Thread watcherThread = ThreadDeathWatcher.watcherThread;
-        if (watcherThread != null) {
-            watcherThread.join(unit.toMillis(timeout));
-            return !watcherThread.isAlive();
+        Thread t = watcherThread.get();
+        if (t != null) {
```

Comment:
IMHO the existing behavior is a bug. The javadocs state the expected behavior of the interface, but the code fails to enforce the behavior. My vote is just to fix the code and remove the race condition.